### PR TITLE
Describe Date Range Parameters

### PR DESCRIPTION
**Pull request by serge-andreyev**

### CODE REVIEW

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters.md [hunk 1]
```diff
@@ -4,20 +4,24 @@ author: Anna Gubareva
...
 
 * [Request and Pass Report Parameter Values](use-report-parameters/request-and-pass-report-parameter-values.md)
-	
-	Lists the standard editors available for submitting report parameter values in a Print Preview.
 
-* [Use Query Parameters](use-report-parameters/use-query-parameters.md)
-	
-	Explains how to link report parameters with query parameters defined in your report's data source.
\ No newline at end of file
+	Illustrates how to assign the default and custom values to a report's parameters and describes the editors that are used to request these values in a Print Preview.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters.md]:
```suggestion
	Illustrates how to assign default and custom values to a report's parameters and describes the editors that are used to request these values in a Print Preview.
```

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters.md [hunk 2]
```diff
@@ -4,20 +4,24 @@ author: Anna Gubareva
...
+
+* [Multi-Value and Cascading Report Parameters](use-report-parameters/multi-value-and-cascading-parameters.md)
+
+	Explains how to configure report parameters so that they accept multiple values, and how to filter a parameter's values based on another parameter's value.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters.md]:
```suggestion
	Explains how to configure report parameters to accept multiple values and filter a parameter's values based on another parameter's value.
```

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters.md [hunk 3]
```diff
@@ -4,20 +4,24 @@ author: Anna Gubareva
...
+
+* [Date Range Parameters](use-report-parameters/date-range-parameters.md)
+
+	Describes how to create date range parameters to filter report data by a specific time period.
+
+* [Query Parameters](use-report-parameters/query-parameters.md)
+
+	Details how to link report parameters with query parameters defined in your report's data source.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters.md]:
```suggestion
	Explains how to link report parameters to query parameters defined in your report's data source.
```

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/query-parameters.md [hunk 4]
```diff
@@ -1,8 +1,8 @@
 ---
-title: Use Query Parameters
+title: Query Parameters
 author: Anna Gubareva
 ---
-# Use Query Parameters
+# Query Parameters
 
 This document provides general information on query parameters and describes common ways of utilizing parametrized SQL queries to filter data at the data source level.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/query-parameters.md]:
```suggestion
This document provides information on query parameters and describes how to use parameterized SQL queries to filter data at data source level.
```

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md [hunk 5]
```diff
@@ -4,75 +4,95 @@ author: Anna Gubareva
...
 	
-	When using [mail merge](../../bind-to-data/use-embedded-fields-mail-merge.md), you can refer to a parameter by adding a question mark before its name.
+	You can refer to a parameter in [mail merge](../../bind-to-data/use-embedded-fields-mail-merge.md). Add a question mark in front of the parameter's name.
 
 	![](../../../../../images/eurd-win-mailmerge-parameters.png)
 
 * **Specify Expressions**
     
-    Parameters can be used as part of [expressions](../../use-expressions.md). To refer to a report parameter, use a question mark before its name.
+    You can include parameters into [expressions](../../use-expressions.md). Use a question mark (?) in front of a parameter's name.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md]:
```suggestion
    Use a question mark (?) in front of a parameter's name to include it in an [expression](../../use-expressions.md).
```

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md [hunk 6]
```diff
@@ -4,75 +4,95 @@ author: Anna Gubareva
...
 
 * **Bind to Data**
 	
-	You can bind a report control to a parameter and display its value in the report. To create a new label bound to a parameter, drag the parameter from the [Field List](../../report-designer-tools/ui-panels/field-list.md) and drop it onto the required band.
+	You can bind a report control to a parameter and display its value in a report. To create a new [label](../../use-report-elements/use-basic-report-controls/label.md) bound to a parameter, drag the parameter from the [Field List](../../report-designer-tools/ui-panels/field-list.md) and drop it onto a band.
 	
 	![](../../../../../images/eurd-win-parameters-for-data-binding.png)
 	
-	When using [mail merge](../../bind-to-data/use-embedded-fields-mail-merge.md), you can refer to a parameter by adding a question mark before its name.
+	You can refer to a parameter in [mail merge](../../bind-to-data/use-embedded-fields-mail-merge.md). Add a question mark in front of the parameter's name.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md]: You can... You can... You can... You can... You are creating a pattern here.

Comment (dirk-pieterse) [reply interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md]: These sentences can be combined.

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md [hunk 7]
```diff
@@ -4,75 +4,95 @@ author: Anna Gubareva
...
-		Switch to this tab to specify a static list of possible values. Each value should have a description that is displayed in a Print Preview.
+		Switch to this tab to specify a static list of values. Each value should have a description that is displayed in Print Preview.
 		
 		![](../../../../../images/eurd-win-parameters-static-values.png)
 
-* **Allow multiple values** (corresponds to the parameter's **Multi-Value** property) - when this option is enabled, a parameter can be assigned a [collection of values](create-multi-value-and-cascading-parameters.md).
+## Use Parameters
 
-* **Allow null value** (corresponds to the parameter's **Allow Null** property) - indicates whether the parameter's value can be unspecified.
+Report parameters pass data to a report before it is generated in Print Preview. You can specify parameter values.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md]: "You can specify parameter value" This is out of context.

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md [hunk 8]
```diff
@@ -4,75 +4,95 @@ author: Anna Gubareva
...
-* **Show in the parameters panel** (corresponds to the parameter's **Visible** property) - enable this option to request the parameter value in a Print Preview. Otherwise, the default parameter value is silently passed to the report.
-* **Supports the collection of standard values** - you can enable this option if the parameter is visible (i.e., its value should be requested in a Print Preview). In this case, you can choose a value from a predefined list. You can either manually populate this list with possible values, or specify a data source from where these values should be obtained.
-	
 	* **Dynamic values**
-		
-		On this tab, you can specify a data source, data adapter (if required) and data member storing parameter values. The value member defines a data field that will provide values to the parameter. The display member defines a data field storing values displayed in a Print Preview.
+
+		Specify a data source, data adapter and data member for the parameter values storage. **Value Member** defines a data field that provides values to the parameter. **Display Member** defines a data field that stores values displayed in Print Preview.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md]:
```suggestion
		Specify a data source, data adapter, and data member for the parameter values storage. **Value Member** defines the data field that provides the parameter's values. **Display Member** defines the data field that stores values displayed in Print Preview.
```

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md [hunk 9]
```diff
@@ -4,75 +4,95 @@ author: Anna Gubareva
...
+
+	![](../../../../../images/eurd-win-parameters-reset-parameter-value.png)
+
+* **Expression**  
+	Specifies an [expression](../../use-expressions.md) that defines the parameter's value based on specific conditions. Click the **Expression** property's ellipsis button and construct an expression in the invoked **Expression Editor**.
+
+	![](../../../../../images/eurd-win-parameters-construct-expression.png)
+
+* **Show in the parameters panel** (corresponds to the parameter's **Visible** property)  
+	Enable this option to request the parameter value in Print Preview. Otherwise, the report assumes the default parameter value.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md]: "the report assumes" Why 'assumes'?

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md [hunk 10]
```diff
@@ -4,75 +4,95 @@ author: Anna Gubareva
...
+* **Allow multiple values** (corresponds to the parameter's **Multi-Value** property)  
+	Enable this option to allow a parameter to accept a [collection of values](create-multi-value-and-cascading-parameters.md).
+* **Allow null value** (corresponds to the parameter's **Allow Null** property)  
+	Enable this option if the parameter's value can be unspecified.
+* **Range Value**  
+	Enable this option if the parameter should specify a range with a start value and an end value. This option applies to _Date_-type parameters. See [Date Range Parameters](date-range-parameters.md) for information on how to configure a date range parameter.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md]:
```suggestion
	Enable this option if the parameter should specify a range with a start and end value. This option applies to _Date_-type parameters. See [Date Range Parameters](date-range-parameters.md) for information on how to configure a date range parameter.
```

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/multi-value-and-cascading-parameters.md [hunk 11]
```diff
@@ -1,8 +1,8 @@
 ---
-title: Create Multi-Value and Cascading Parameters
+title: Multi-Value and Cascading Parameters
 author: Anna Gubareva
 ---
-# Create Multi-Value and Cascading Parameters
+# Multi-Value and Cascading Parameters
 
 This document describes the implementation of multi-value and cascading parameters. Multi-value parameters can accept more than a single value, and cascading parameters display values corresponding to current values of other parameters.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/multi-value-and-cascading-parameters.md]:
```suggestion
This document describes how to implement multi-value and cascading parameters. Cascading parameters display values that correspond to other parameters' values.
```

**Review by dirk-pieterse — CHANGES_REQUESTED** [review]

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters.md (hunk 3 above)
Comment (serge-andreyev) [reply interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters.md]: Ok, changed

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md (hunk 6 above)
Comment (serge-andreyev) [reply interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md]: Ok

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md (hunk 9 above)
Comment (serge-andreyev) [reply interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md]: Changed to "takes"

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md (hunk 7 above)
Comment (serge-andreyev) [reply interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md]: Changed

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md (hunk 10 above)
Comment (serge-andreyev) [reply interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md]: Ok

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/multi-value-and-cascading-parameters.md (hunk 11 above)
Comment (serge-andreyev) [reply interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/multi-value-and-cascading-parameters.md]: Ok

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/date-range-parameters.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+1. Specify the options below in the invoked **Add New Parameter** dialog and click **OK**.
+
+    - **Name** - the parameter's name;
+    - **Description** - the parameter's description displayed in Print Preview's **Parameters** panel;
+    - **Type** - the parameter's data type. Set it to **Date** to create a date range;
+    - **Range Value** - enable this option to create a date range.
+
+    ![](../../../../../images/eurd-win-use-date-ranges-design-add-param-dialog.png)
+
+1. The _dateRange_ parameter appears in the **Field List** and includes the inner _dateRange_Start_ and _dateRange_End_ parameters. Select the _dateRange_ parameter and switch to the **Property Grid** to specify the inner parameters' default values.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/date-range-parameters.md]: What are "the inner parameters"?

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/date-range-parameters.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+
+    ![](../../../../../images/eurd-win-use-date-ranges-design-configure-subparam.png)
+
+    The **Value** property allows you to specify a static default value. Provide an expression for the **Value** property to specify a dynamic default value. Click the **Value** property's marker, select **Value Expression** and specify an expression in the invoked editor.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/date-range-parameters.md]:
```suggestion
    The **Value** property allows you to specify a static value. To do this. provide an expression for the **Value** property which specifies a dynamic value. Click the **Value** property's marker, select **Value Expression**, and specify an expression in the invoked editor.
```

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/date-range-parameters.md [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+
+    ![](../../../../../images/eurd-win-use-date-ranges-design-value-expression.png)
+
+1. Use the start and end parameter names in the report's filter string to [filter data](../filter-data/filter-data-at-the-report-level.md) by the created date range. Select the report, click the **Filter String** property's ellipsis button in the **Property Grid** and construct a filter string in the invoked **FilterString Editor**.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/date-range-parameters.md]:
```suggestion
1. Use the start and end parameter names in the report's filter string to [filter data](../filter-data/filter-data-at-the-report-level.md) by the specified date range. Select the report, click the **Filter String** property's ellipsis button in the **Property Grid**, and construct a filter string in the invoked **FilterString Editor**.
```

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/query-parameters.md [hunk 15]
```diff
@@ -1,10 +1,10 @@
...
 author: Anna Gubareva
 ---
-# Use Query Parameters
+# Query Parameters
 
-This document provides general information on query parameters and describes common ways of utilizing parametrized SQL queries to filter data at the data source level.
+This document provides information on query parameters and describes how to use parameterized SQL queries to filter data at data source level.
 
 ## <a name="overview"></a>Query Parameters Overview
 A query parameter holds an external value that is inserted into an SQL statement before query execution. This value can be either static or dynamically generated by an associated expression.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/query-parameters.md]:
```suggestion
A query parameter holds an external value that is inserted into an SQL statement before query execution. This value can be static or an associated expression can generate it dynamically.
```

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md [hunk 16]
```diff
@@ -4,75 +4,95 @@ author: Anna Gubareva
...
-		Switch to this tab to specify a static list of possible values. Each value should have a description that is displayed in a Print Preview.
+		Switch to this tab to specify a static list of values. Each value should have a description that is displayed in Print Preview.
 		
 		![](../../../../../images/eurd-win-parameters-static-values.png)
 
-* **Allow multiple values** (corresponds to the parameter's **Multi-Value** property) - when this option is enabled, a parameter can be assigned a [collection of values](create-multi-value-and-cascading-parameters.md).
+## Use Parameters
 
-* **Allow null value** (corresponds to the parameter's **Allow Null** property) - indicates whether the parameter's value can be unspecified.
+Report parameters pass data to a report before it is generated in Print Preview. You can specify these parameter's values.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md]:
```suggestion
Report parameters pass data to a report before it is generated in Print Preview. You can specify these parameter values.
```

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/multi-value-and-cascading-parameters.md [hunk 17]
```diff
@@ -1,10 +1,10 @@
 ---
-title: Create Multi-Value and Cascading Parameters
+title: Multi-Value and Cascading Parameters
 author: Anna Gubareva
 ---
-# Create Multi-Value and Cascading Parameters
+# Multi-Value and Cascading Parameters
 
-This document describes the implementation of multi-value and cascading parameters. Multi-value parameters can accept more than a single value, and cascading parameters display values corresponding to current values of other parameters.
+This document describes how to implement multi-value and cascading parameters. Cascading parameters display values that correspond to other parameters'.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/multi-value-and-cascading-parameters.md]: Other parameters' what?

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md [hunk 18]
```diff
@@ -4,75 +4,95 @@ author: Anna Gubareva
...
+* **Show in the parameters panel** (corresponds to the parameter's **Visible** property)  
+	Enable this option to request the parameter value in Print Preview. Otherwise, the report takes the default parameter value.
+* **Allow multiple values** (corresponds to the parameter's **Multi-Value** property)  
+	Enable this option to allow a parameter to accept a [collection of values](create-multi-value-and-cascading-parameters.md).
+* **Allow null value** (corresponds to the parameter's **Allow Null** property)  
+	Enable this option if the parameter's value can be unspecified.
+* **Range Value**  
+	Enable this option if the parameter should specify a range with a start and end value. This option applies to _Date_-type parameters. See [Date Range Parameters](date-range-parameters.md) for information on how to configure a date range parameter.
+* **Supports the collection of standard values**  
+	Applies if the parameter is visible (its value is requested in Print Preview). You can choose a value from a predefined list. You can populate this list with static values, or specify a data source from where the values are obtained.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md]:
```suggestion
	Applies if the parameter is visible (its value is requested in Print Preview). You can choose a value from a predefined list which is populated with static values, or specify a data source from where the values are obtained.
```

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md [hunk 19]
```diff
@@ -4,75 +4,95 @@ author: Anna Gubareva
...
 	* **Dynamic values**
-		
-		On this tab, you can specify a data source, data adapter (if required) and data member storing parameter values. The value member defines a data field that will provide values to the parameter. The display member defines a data field storing values displayed in a Print Preview.
+
+		Specify a data source, data adapter, and data member for the parameter values storage. **Value Member** defines the data field that provides the parameter's values. **Display Member** defines the data field that stores values displayed in Print Preview.
 		
 		![](../../../../../images/eurd-win-parameters-dynamic-values.png)
 		
-		The value type of the specified data member should match the specified parameter type.
+		The specified data member's value type should match the specified parameter type.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md]:
```suggestion
		The data member's value type should match the parameter type.
```

**Review by dirk-pieterse — CHANGES_REQUESTED** [review]

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/date-range-parameters.md (hunk 12 above)
Comment (serge-andreyev) [reply interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/date-range-parameters.md]: Removed the word 'inner'

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/multi-value-and-cascading-parameters.md (hunk 17 above)
Comment (serge-andreyev) [reply interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/multi-value-and-cascading-parameters.md]: ...other parameters' values. Added to the doc

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md (hunk 19 above)
Comment (serge-andreyev) [reply interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/parameters-overview.md]: I'm getting rid of the word 'should'. So I changed that to:
If the data member's value type does not match the parameter type, the validation rejects the value.

File: interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/date-range-parameters.md (hunk 13 above)
Comment (serge-andreyev) [reply interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/shape-report-data/use-report-parameters/date-range-parameters.md]: Changed to:
    The **Value** property allows you to specify a static value. To specify a dynamic value, click the **Value** property's marker, select **Value Expression** and specify an expression in the invoked editor.

**Pull request merged**